packages/stream-json/reader.js

```diff
@@ -0,0 +1,26 @@
+// @ts-check
+/// <reference types="ses"/>
+
+/* Adapts a Reader<Uint8Array> to a JSON object Reader<any> where reader and
+ * writer streeams are modeled as hybrid async iterators + generators.
+ */
+const textDecoder = new TextDecoder();
+
+/**
+ * @param {import('@endo/stream').Reader<Uint8Array, undefined>} reader
+ */
+async function* makeJsonAsyncIterator(reader) {
+  for await (const bytes of reader) {
+    const text = textDecoder.decode(bytes);
+    yield JSON.parse(text);
+  }
+}
```

Comment:
It is a shame we need to leave Jessie. Perhaps we can come up with a design rule for restricted async generators like we have for async functions. For example, if the `yield` is always at the top level of the body of an top level for-await-of, that would cover this case, probably many other related cases, and would still admit a simple implementation strategy short of CPS transforming.

Reply:
We certainly could implement this particular case with the iterator interface directly and avoid the need, but there are certainly other kinds of adapters that benefit from async generators.